server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -178,9 +182,11 @@ public RemoteSegmentMetadata readLatestMetadataFile() throws IOException {
 
         List<String> metadataFiles = remoteMetadataDirectory.listFilesByPrefixInLexicographicOrder(
             MetadataFilenameUtils.METADATA_PREFIX,
-            1
+            METADATA_FILES_TO_FETCH
         );
 
+        verifyNoMultipleWriters(metadataFiles);
```

Comment:
What would be the overhead of this method? `readLatestMetadataFile` is called during each replication call. Do we need to check conflicts that frequently? Instead, can we verify as part of `deleteStaleSegments(int lastNMetadataFilesToKeep)`?

Reply:
The overhead should be minimal. It is just reading the names of 100 files (in happy cases just ~10) and tokenizing it . `deleteStaleSegments` is completely async and we don't wait it to finish in `RemoteStoreRefreshListener` .

Reply:
We call `readLatestMetadataFile` on each refresh on each replica. If there are more than 1 replicas, we may not want to perform the same check on multiple nodes.

Reply:
I did try doing `failShard` in `deleteStateSegments` . However the problem there is once the primary fails, replicas takeover and the cluster becomes green again .